facia/app/services/dotcomrendering/FaciaPicker.scala

```diff
@@ -172,10 +96,13 @@ object FaciaPicker extends GuLogging {
     if (isRss) LocalRender
     else if (forceDCROff) LocalRender
     else if (forceDCR) RemoteRender
-    else if (isNetworkFront)
-      if (dcrCouldRender && isInNetworkFrontTest) RemoteRender else LocalRender
-    else if (dcrCouldRender && dcrSwitchEnabled) RemoteRender
-    else LocalRender
+    else if (dcrCouldRender && dcrSwitchEnabled) {
+      isNetworkFront match {
+        case false                        => RemoteRender
+        case true if isInNetworkFrontTest => RemoteRender
+        case _                            => LocalRender
```

Comment:
Not a blocker, but because theres only 1 unhandled branch you could be explicit here

```suggestion
        case true if !isInNetworkFrontTest        => LocalRender
```

Reply:
Tru, I guess that's the benefit of having an case matcher: explicit and exhaustive!

Reply:
Unfortunately, no, the compiler isn’t able to narrow it down for us:

<img width="414" alt="image" src="https://github.com/guardian/frontend/assets/76776/3a354e0d-0f0f-4aba-9086-8b06761f7d73">